src/mol-plugin-state/manager/focus-camera/orient-axes.ts

```diff
@@ -0,0 +1,247 @@
+ * If `referenceRotation` is provided, select the one nearest to `referenceRotation`.
+ * Otherwise use arbitrary rules to ensure the orientation after transform does not depend on the original orientation.
+ */
+export function layingTransform(coords: number[], referenceRotation?: Mat3): { rotation: Mat3, origin: Vec3 } {
+    if (coords.length === 0) {
+        console.warn('Skipping PCA, no atoms');
+        return { rotation: Mat3.identity(), origin: Vec3.zero() };
+    }
+    const axes = PrincipalAxes.calculateMomentsAxes(coords);
+    const normAxes = PrincipalAxes.calculateNormalizedAxes(axes);
+    const R = mat3FromRows(normAxes.dirA, normAxes.dirB, normAxes.dirC);
+    avoidMirrorRotation(R); // The SVD implementation seems to always provide proper rotation, but just to be sure
+    const flip = referenceRotation ? minimalFlip(R, referenceRotation) : canonicalFlip(coords, R, axes.origin);
+    Mat3.mul(R, flip, R);
+    return { rotation: R, origin: normAxes.origin };
+}
+
+/** Try these selection strategies until having at least `minAtoms` atoms:
+ * 1. only "polymer" atoms (e.g. C-alpha and O3')
+ * 2. all non-hydrogen atoms with exception of water (HOH)
+ * 3. all atoms
+ * Return the coordinates in a flattened array (in triples).
+ * If the total number of atoms is less than `minAtoms`, return only those. */
+function selectCoords(structs: Structure[], minAtoms: number): number[] {
+    let coords = selectCACoords(structs);
+    if (coords.length >= 3 * minAtoms) return coords;
+
+    coords = selectHeavyCoords(structs);
+    if (coords.length >= 3 * minAtoms) return coords;
+
+    coords = selectAllCoords(structs);
+    return coords;
+}
+
+/** Select coordinates of C-alpha and O3' atoms */
+function selectCACoords(structs: Structure[]): number[] {
+    const coords: number[] = [];
+    for (const struct of structs) {
+        for (const unit of struct.units) {
+            const { x, y, z } = unit.conformation;
+            for (let i = 0; i < unit.polymerElements.length; i++) {
+                const index = unit.polymerElements[i];
+                coords.push(x(index), y(index), z(index));
+            }
+        }
+    }
+    return coords;
+}
+
+/** Select coordinates of non-hydrogen atoms, excluding water */
+function selectHeavyCoords(structs: Structure[]): number[] {
+    const coords: number[] = [];
+    for (const struct of structs) {
+        for (const unit of struct.units) {
+            const { x, y, z } = unit.model.atomicConformation;
+            for (let i = 0; i < unit.elements.length; i++) {
+                const index = unit.elements[i];
+                const compound = unit.model.atomicHierarchy.atoms.label_comp_id.value(index);
+                const element = unit.model.atomicHierarchy.atoms.type_symbol.value(index);
+                if (element !== 'H' && compound !== 'HOH') {
+                    coords.push(x[index], y[index], z[index]);
+                }
+            }
+        }
+    }
```

Comment:
Usually we use this pattern for accessing the props:

```ts
const loc = StructureElement.Location.create(structure);
const type_symbol = StructureProperties.atom.type_symbol;
for (const unit of units) {
  const { elements } = unit;
  loc.unit = unit;
  for (let i = 0; i < unit.elements.length; i++)
    loc.element = elements[i];
    const ts = type_symbol(loc);
  }
}
```

Can you update your code with that please?

Reply:
Done, and merged the three similar function into one function with options.